### PR TITLE
Autogenstudio - Add GroupChat Support to UI

### PR DESCRIPTION
**Pull request by victordibia**
## Why are these changes needed?

Add support for GroupChat, and other quality of life improvements to autogenstudio

https://github.com/microsoft/autogen/assets/1547007/58e66ce1-3c10-4c5e-b45d-783121d151ec

## Related issue number

#1285  #1265 #1294 #1296 

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by gagb** [comment]
@victordibia are you able to build this PR in the studio devcontainer and get `autogenstudio ui` to succeed?

**Comment by victordibia** [comment]
@gagb , I just tried this and I am able to run the studio devcontainer. and `autogenstudio ui` runs once the devcontainer loads. Are you seeing any issues?

**Review by reconsumeralization — APPROVED** [review]
Looks good on my end.

File: website/blog/2023-12-01-AutoGenStudio/index.mdx [hunk 1]
```diff
@@ -189,6 +191,21 @@ As we continue to develop and refine AutoGen Studio, the road map below outlines
...
+
+We welcome contributions to AutoGen Studio. We recommend the following general steps to contribute to the project:
+
+- Review the overall AutoGen project [contribution guide](https://github.com/microsoft/autogen?tab=readme-ov-file#contributing)
+- Please review the AutoGen Studio [roadmap](https://github.com/microsoft/autogen/issues/737) to get a sense of the current priorities for the project. Help is appreciated especially with Studio issues tagged with `help-wanted`
+- Please initiate a discussion on the roadmap issue or a new issue to discuss your proposed contribution.
+- Please review the autogenstudio dev branch here [dev branch](https://github.com/microsoft/autogen/tree/autogenstudio) and use as a base for your contribution. This way, your contribution will be aligned with the latest changes in the AutoGen Studio project.
+- Submit a pull request with your contribution!
+- If you are modifying AutoGen Studio, it has its own devcontainer. See instructions in `.devcontainer/README.md` to use it
+- Please use the tag `studio` for any issues, questions, and PRs related to Studio
```
Comment (sonichi) [inline website/blog/2023-12-01-AutoGenStudio/index.mdx:206]:
```suggestion
- Review the overall AutoGen project [contribution guide](https://github.com/microsoft/autogen?tab=readme-ov-file#contributing).
- Please review the AutoGen Studio [roadmap](https://github.com/microsoft/autogen/issues/737) to get a sense of the current priorities for the project. Help is appreciated especially with Studio issues tagged with `help-wanted`.
- Please initiate a discussion on the roadmap issue or a new issue to discuss your proposed contribution.
- Please review the autogenstudio dev branch here [dev branch].(https://github.com/microsoft/autogen/tree/autogenstudio) and use as a base for your contribution. This way, your contribution will be aligned with the latest changes in the AutoGen Studio project.
- Submit a pull request with your contribution!
- If you are modifying AutoGen Studio in vscode, it has its own devcontainer. See instructions in `.devcontainer/README.md` to use it.
- Please use the tag `studio` for any issues, questions, and PRs related to Studio.
```

File: website/blog/2023-12-01-AutoGenStudio/index.mdx [hunk 2]
```diff
@@ -203,5 +220,9 @@ A: Yes, you can view the generated messages in the debug console of the web UI,
 **Q: Where can I find documentation and support for AutoGen Studio?**
 A: We are constantly working to improve AutoGen Studio. For the latest updates, please refer to the [AutoGen Studio Readme](https://github.com/microsoft/autogen/tree/main/samples/apps/autogen-studio). For additional support, please open an issue on [GitHub](https://github.com/microsoft/autogen) or ask questions on [Discord](https://discord.gg/pAbnFJrkgZ).
 
+**Q: Can I use Other Models with AutoGen Studio?**
+Yes. AutoGen standardizes on the openai model api format, and you can use any api server that offers an openai compliant endpoint. In the AutoGen Studio UI, each agent has an `llm_config` field where you can input your model endpoint details including `model name`, `api key`, `base url`, `model type` and `api version`. For Azure OpenAI models, you can find these details in the Azure portal. Note that for Azure OpenAI, the `model name` is the deployment id or engine, and the `model type` is "azure".
```
Comment (sonichi) [inline website/blog/2023-12-01-AutoGenStudio/index.mdx:224]: model type -> api type

File: website/blog/2023-12-01-AutoGenStudio/index.mdx (hunk 1 above)
Comment (gagb) [reply website/blog/2023-12-01-AutoGenStudio/index.mdx:206]: If you add periods here add it everywhere. Including README. I personally find periods for enumerating large number of short bullets unnecessary but consistency is more important than anything else.

**Comment by gagb** [comment]
> @gagb , I just tried this and I am able to run the studio devcontainer. and `autogenstudio ui` runs once the devcontainer loads. Are you seeing any issues?

I haven't tried today but when I tested yesterday, in the studio container the autogenstudio ui command runs, but once I try to interact with the webpage (which it opens via a port redirect) the webpage encounters many error messages (toasts). I will have time to test again tomorrow morning. Did that correctly work for you?

**Comment by victordibia** [comment]
> > @gagb , I just tried this and I am able to run the studio devcontainer. and `autogenstudio ui` runs once the devcontainer loads. Are you seeing any issues?
> 
> I haven't tried today but when I tested yesterday, in the studio container the autogenstudio ui command runs, but once I try to interact with the webpage (which it opens via a port redirect) the webpage encounters many error messages (toasts). I will have time to test again tomorrow morning. Did that correctly work for you?

Thanks. I'll test again, but If you can share the exact toast errors (e.g., if it is an api key thing .. ), it would be helpful

**Comment by gagb** [comment]
> > > @gagb , I just tried this and I am able to run the studio devcontainer. and `autogenstudio ui` runs once the devcontainer loads. Are you seeing any issues?
> > 
> > 
> > I haven't tried today but when I tested yesterday, in the studio container the autogenstudio ui command runs, but once I try to interact with the webpage (which it opens via a port redirect) the webpage encounters many error messages (toasts). I will have time to test again tomorrow morning. Did that correctly work for you?
> 
> Thanks. I'll test again, but If you can share the exact toast errors (e.g., if it is an api key thing .. ), it would be helpful

I am no longer seeing the toast errors! I am not sure what changed.

But I am still seeing interactions that feel buggy for creating a simple chatgpt like experience -- I can meet 1-1 and show. Some of these are related to adding default params models and agents that do not exist. Why is min max consecutive replies set to 2?

Would be worth addressing these in a separate PR, ideally, before making a release.

**Review by gagb — APPROVED** [review]

**Pull request merged**